imap/sievedir.c

```diff
@@ -87,11 +87,15 @@ EXPORTED int sievedir_foreach(const char *sievedir,
 
     while ((dir = readdir(dp)) != NULL) {
         const char *name = dir->d_name;
+        const char *dot;
         char target[PATH_MAX] = "";
         struct stat sbuf;
 
         if (!strcmp(name, ".") || !strcmp(name, "..")) continue;
 
+        /* ignore transient .NEW files */
+        if ((dot = strrchr(name, '.')) && !strcmp(dot, ".NEW")) continue;
```

Comment:
In theory this would match foo.NEW.bar.  We have to do enough comparisons of string suffixes that we should probably create a strrncmp() or something